[Docs] Enhance Documentation: Add Quiz Part for Academy Course in new page

Vincamine opened the pull request:
**Notes for Reviewers**

This PR fixes #729 




**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

l5io commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+---
+title: Enriching Course With Activities
```
Enriching Courses - plural, perhaps?

l5io commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+
+The [Layer5 Cloud Academy](https://cloud.layer5.io/academy/content) courses are structured as modular learning experiences. 
+
+Each **Course** is broken down into **Modules**, and each Module consists of learning activities. As a content developer, you can enrich your modules with three types of learning activities:
```
If we have definitions of Courses and Modules, link to them, please.

leecalcote commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+
+Each **Course** is broken down into **Modules**, and each Module consists of learning activities. As a content developer, you can enrich your modules with three types of learning activities:
+
+- **Pages** – Text-based content for theoretical learning  
+- **Quizzes** – Knowledge checks for concept reinforcement  
+- **Labs** – Hands-on practical exercises  
```
Link to their definitions, if we have them.

leecalcote commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+
+### File Structure
+
+Quizzes are placed inside a `knowledge-check/` directory under each module:
```
Is the name of the folder mandatorily `knowledge-check`, @vr-varad @Namanv0509 @zihanKuang ?

leecalcote commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
````diff
@@ -0,0 +1,205 @@
+
+```yaml
+---
+title: "Knowledge Check"
````
Be aware: the `title` property is likely to disappear. It adds no value.

vr-varad commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+course-name/
+└── module-1/
+└── module-2/
+│   └── knowledge-check/
```
The file structure is updated, it's like each module has a quiz.md no knowledge-check folder

ShashaankS commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
```diff
@@ -0,0 +1,205 @@
+### Frequently Asked Questions
+1. **Must the quiz file be named exactly `quiz.md?`?**
+
+    Yes, the filename must be quiz.md for the system to recognize it.
```
Its not mandetory to use quiz.md,we can use `<any-name>.md` with metadata of quiz type.
To be specific it should have:
```
layout: "quiz"
type: "quiz"
```
in meta data

Vincamine commented:
Update: 
1.added key rules:
1.1 Each level can have a quiz.md file, and passing the quiz marks that level as completed.(with examples)
1.2 Quizzes can have prerequisites that must be completed before taking the quiz.
1.3. add Quiz Quantity Per Level rules
1.4 Unified Quiz Structure

2.adjusted other related part.
3.added the part link for (page, quiz and lab)

zihanKuang commented on content/en/cloud/academy/enriching-course-with-activities/index.md:
````diff
@@ -0,0 +1,245 @@
+            └── module-1/
+                └── quiz.md
+```
+
````
Regarding Aabid's statement: "in terms of no of test at a level currently the system supports multiple tests at module and single test at course and learning path level. for proper tracking of completion and progress it is need to have a test defined at each course and learning path / challenge / or in general top most level."

Add and imporve this explanation using an alert box or as a direct section within the documentation.

Namanv0509 commented:
<img width="1050" height="508" alt="image" src="https://github.com/user-attachments/assets/b3c9f28d-52eb-46d6-9a9c-2aaafe0e4120" />


i think it needs the `type:quiz`

Vincamine commented:
> The title and first paragraph need to chaneg. We can add quizzes at all levels of content hierarchy, so those sections will need to be updated accordingly.
> 
> Also, if we're not planning to immediately write about "page" and "lab" types, the title should simply be "Creating Quizzes", and the content should focus solely on quizzes.

creating "pages" looks like a necessity? I'll use more quizzes-focused title.

leecalcote commented:
My, my, my, don't you have a popular PR, @Vincamine? 😄

zihanKuang commented on content/en/cloud/academy/integrating-assessments-in-the-academy/index.md:
```diff
@@ -0,0 +1,328 @@
+
+## Frequently Asked Questions
+
+<details style="margin-bottom: 1em;">
```
Looks great! Just a small but important formatting tip for this FAQ section using the `<details>` tag.

You can use Markdown for the content inside a `<summary>` block, but for it to render correctly, there must be a blank line after the closing `</summary>` tag.

This is because most Markdown parsers will treat everything as a single raw HTML block if there's no empty line. That blank line acts as a crucial separator, telling the parser. You can test this yourself

**Example:**

```html
<details>
  <summary>Why is a blank line needed?</summary>

  **This text** will be properly rendered in bold.
</details>

<details>
  <summary>Why is a blank line needed?</summary>
  **This text** will literally show the ** **.
</details>
```
